addon-resizer/nanny/kubernetes_client.go

```diff
@@ -53,6 +72,45 @@ func (k *kubernetesClient) CountNodes() (uint64, error) {
 	return uint64(len(result.Items)), err
 }
 
+func (k *kubernetesClient) countNodesThroughMetrics() (uint64, error) {
+	rawMetrics, err := k.clientset.Core().RESTClient().Get().RequestURI("/metrics").DoRaw()
+	if err != nil {
+		return 0, err
+	}
+
+	decoder := expfmt.SampleDecoder{
```

Comment:
The code would be cleaner if you used `expfmt.TextParser.` You'd get a map from metric name to `dto.MetricFamily`, then you would have to just iterate over metrics to find sample with proper label.

Reply:
I don't really see the difference - the code would be almost exactly the same - I just wouldn't need the catching of io.EOF.

Given that this is what is used in couple different places in both k/k and k/perf-tests (thus is validated approach) and TextParser isn't used anywhere, I'm going to stick to what there is now.

Reply:
I see a huge difference between your code and this snippet

```
metrics := expfmt.TextParser{}.TextToMetricFamilies(rawMetrics)
if metricFamily, ok := metrics[objectCountMetricName]; ok {
   for _, m := metricFamily.Metric {
      for _, l := m.Label {
        if l.Name == "resource" && l.Value = nodeResourceName {
           value := uint64(m.Counter.Value)
		   if value < 0 {
		     return 0, fmt.Errorf("metric unknown")
		   }
		   return value, nil
        }
      }
   }
}
return 0, fmt.Errorf("metric unset")
```

Reply:
TBH, I see bigger value in consistency with number of other places, than in saving 5 lines of simple code.

Reply:
If it's about consistency then `Decoder` with `FmtText` format calls `TextToMetricFamilies` underneath:

https://github.com/prometheus/common/blob/d0f7cd64bda49e08b22ae8a730aa57aa0db125d6/expfmt/decode.go#L127

It's not only about saving lines of code, it's also about readability.  Libraries evolve, copying-pasting solutions for sake of consistency is not always optimal.

That said, I don't see a point in arguing about a few lines of code. Your code is correct, I'd do it differently, but it looks good to me.